docs/DeveloperGuide.md

```diff
@@ -6,33 +6,183 @@
 
```

Comment:
can remove if not used

![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/61624545/e53adc23-c3a3-490a-948b-90c8b3521be1)